### PR TITLE
Fix Private Registry broken link and images

### PR DESCRIPTION
**Pull request by laurapacilio**
This PR fixes a broken anchor that was changed in a previous PR. It also updates the images on several of the Private Module Registry pages because the main TFC navigation bar now allows users to get to the PMR through "Registry" not "Module".

I have
- [x] Verified that this builds locally.
- [x] Assigned at least one label.
- [x] Assigned at least one reviewer.

### CODE REVIEW

File: content/source/docs/cloud/registry/design.html.md [hunk 1]
```diff
@@ -23,15 +23,13 @@ Navigate to the modules list with the top-level "Modules" button, then click the
 
 ![Terraform Cloud screenshot: the design configuration button](./images/design-start.png)
 
-This brings you to the "Select Modules" page.
+The Select Modules page appears.
 
 ## Adding Modules
 
-![Terraform Cloud screenshot: the select modules page](./images/design-add-module.png)
```
Comment (judithpatudith) [inline content/source/docs/cloud/registry/design.html.md:30]: It feels a little weird not to have a screen shot here. Could we update or crop the existing one?

File: content/source/docs/cloud/registry/using.html.md [hunk 2]
```diff
@@ -36,35 +36,23 @@ The module page appears, containing a list of all available modules. You can fil
...
 
 ![Terraform Enterprise screenshot: shared module](./images/using-modules-list-shared.png)
 
-Modules in your current organization that are shared with other organizations have a badge that says "Sharing".
-
-![Terraform Enterprise screenshot: sharing module](./images/using-modules-list-sharing.png)
 
 ### Viewing Module Details and Versions
 
-Click a module's "Details" button to view its details page. Use the "Versions" dropdown in the upper right to switch between the available versions, and use the Readme/Inputs/Outputs/Dependencies/Resources tabs to view detailed documentation and information about a version.
```
Comment (judithpatudith) [inline content/source/docs/cloud/registry/using.html.md:49]: was this not accurate? I still see the detail button on the screen shot?

**Review by judithpatudith — APPROVED** [review]
Looks good

**Review by rberlind — APPROVED** [review]
My review was superficial, but the changes are fine with me.

File: content/source/docs/cloud/registry/design.html.md (hunk 1 above)
Comment (laurapacilio) [reply content/source/docs/cloud/registry/design.html.md:30]: okay fair! I added one back in :-)

File: content/source/docs/cloud/registry/using.html.md (hunk 2 above)
Comment (laurapacilio) [reply content/source/docs/cloud/registry/using.html.md:49]: Yeah when I go into TFC and I go to Registry and want to look at the available modules, there's no details button when I mouse over. I just mouse over and have to click it and then it goes to the details page. Thank you for flagging that on the shared TFE image!  I got in touch with the TFE team and they gave me a new image from a recent instance that doesn't show the details button (they must have taken that out). I pushed up the new image, so we should be all set!

**Comment by laurapacilio** [comment]
P.S. Ignoring the link checker as the links to learn _do_ work. Marketing Web Dev team confirmed this is okay to do.

**Pull request merged**